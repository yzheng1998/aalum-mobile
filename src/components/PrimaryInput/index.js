import React, { Component } from 'react'
import {
  InputField,
  RoundContainer,
  Icon,
  MainContainer,
  Title,
  ErrorMessage
} from './styles'

class PrimaryInput extends Component {
  focus = () => {
    this.textInput.focus()
  }

  render() {
    const { as, onPress, icon, title, error, ...rest } = this.props
    return (
      <MainContainer>
        {title && <Title error={error}>{title}</Title>}
        <RoundContainer as={as} onPress={onPress} error={error}>
          <InputField
            ref={input => {
              this.textInput = input
            }}
            autoCorrect={false}
            icon={icon}
            autoCapitalize="none"
            {...rest}
          />
          {icon && <Icon>{icon}</Icon>}
        </RoundContainer>
        <ErrorMessage>{error || ''}</ErrorMessage>
      </MainContainer>
    )
  }
}

export default PrimaryInput
