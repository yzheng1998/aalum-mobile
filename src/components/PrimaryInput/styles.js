import styled from 'styled-components'
import { Platform } from 'react-native'
import { themeGet } from 'styled-system'

export const MainContainer = styled.TouchableOpacity`
  margin-left: 32px;
  margin-right: 32px;
  padding-top: 8px;
`

export const Title = styled.Text`
  margin-left: 16px;
  margin-bottom: 10px;
  font-family: ${themeGet('fonts.title')};
  color: ${({ error }) =>
    error ? themeGet('colors.error') : themeGet('colors.charcoal')};
  font-size: 16px;
  line-height: 19px;
  margin-top: 8px;
`

export const InputField = styled.TextInput`
  width: ${({ icon }) => (icon ? '90%' : '100%')};
  font-size: 14px;
  font-family: ${themeGet('fonts.bodyText')};
  color: ${themeGet('colors.charcoal')};
`

export const RoundContainer = styled.View`
  border-width: ${({ error }) => (error ? '2px' : '1.5px')};
  height: 45px;
  border-radius: 30px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: ${Platform.OS === 'ios' ? '8px 16px 8px 16px' : '0px 12px 0px 12px'};
  border-color: ${({ error }) =>
    error ? themeGet('colors.error') : themeGet('colors.grey')};
  background-color: ${themeGet('colors.white')};
`

RoundContainer.displayName = 'RoundContainer'

export const Icon = styled.View`
  align-items: center;
  justify-content: center;
  padding-bottom: 1.5px;
`

export const ErrorMessage = styled.Text`
  margin-top: 5px;
  margin-left: 10px;
  color: ${themeGet('colors.error')};
  font-family: ${themeGet('fonts.title')};
  font-size: 13px;
  line-height: 18px;
`
