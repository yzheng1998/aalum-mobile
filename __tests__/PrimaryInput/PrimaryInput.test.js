import React from 'react'
import PrimaryInput from '../../src/components/PrimaryInput'
import { shallow, mount } from 'enzyme'
import mountWithTheme from '../helpers/theme'

describe('PrimaryInput', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<PrimaryInput />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ title: 'title' })
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ error: 'error' })
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ icon: 'icon' })
    expect(wrapper).toMatchSnapshot()
  })

  test('Round Container handles onPress correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(<PrimaryInput onPress={onPressMock} />)

    const touchable = wrapper.find('RoundContainer')
    touchable.simulate('press')

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })
})
