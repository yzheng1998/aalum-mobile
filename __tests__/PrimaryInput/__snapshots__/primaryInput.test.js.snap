// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryInput renders correctly 1`] = `
<Styled(TouchableOpacity)
  activeOpacity={1}
  onPress={[Function]}
>
  <RoundContainer>
    <Styled(TextInput)
      autoCapitalize="none"
      autoCorrect={false}
    />
  </RoundContainer>
  <Styled(Text) />
</Styled(TouchableOpacity)>
`;

exports[`PrimaryInput renders correctly 2`] = `
<Styled(TouchableOpacity)
  activeOpacity={1}
  onPress={[Function]}
>
  <Styled(Text)>
    title
  </Styled(Text)>
  <RoundContainer>
    <Styled(TextInput)
      autoCapitalize="none"
      autoCorrect={false}
    />
  </RoundContainer>
  <Styled(Text) />
</Styled(TouchableOpacity)>
`;

exports[`PrimaryInput renders correctly 3`] = `
<Styled(TouchableOpacity)
  activeOpacity={1}
  onPress={[Function]}
>
  <Styled(Text)
    error="error"
  >
    title
  </Styled(Text)>
  <RoundContainer
    error="error"
  >
    <Styled(TextInput)
      autoCapitalize="none"
      autoCorrect={false}
    />
  </RoundContainer>
  <Styled(Text)>
    error
  </Styled(Text)>
</Styled(TouchableOpacity)>
`;

exports[`PrimaryInput renders correctly 4`] = `
<Styled(TouchableOpacity)
  activeOpacity={1}
  onPress={[Function]}
>
  <Styled(Text)
    error="error"
  >
    title
  </Styled(Text)>
  <RoundContainer
    error="error"
  >
    <Styled(TextInput)
      autoCapitalize="none"
      autoCorrect={false}
      icon="icon"
    />
    <Styled(View)>
      icon
    </Styled(View)>
  </RoundContainer>
  <Styled(Text)>
    error
  </Styled(Text)>
</Styled(TouchableOpacity)>
`;
